core/src/test/java/kafka/test/api/ShareConsumerTest.java

```diff
@@ -1832,83 +1835,129 @@ public void testShareAutoOffsetResetByDurationInvalidFormat() throws Exception {
+                            ProducerRecord<String, String> record = new ProducerRecord<>(
+                                tpMulti.topic(),
+                                tpMulti.partition(),
+                                null,
+                                key,
+                                "value"
+                            );
+                            try {
+                                producer.send(record);
+                                producer.flush();
+                                // count only correctly produced records
+                                prodState.count().incrementAndGet();
+                                produced.add(key);
+                            } catch (Exception e) {
+                                // ignore
+                            }
+                        }
+                    }
+                }
+            );
 
-            // get current share coordinator node
-            SharePartitionKey key = SharePartitionKey.getInstance(groupId, new TopicIdPartition(topicId, tpMulti));
-            int shareGroupStateTp = Utils.abs(key.asCoordinatorKey().hashCode()) % 3;
-            List<Integer> curShareCoordNodeId = admin.describeTopics(List.of(Topic.SHARE_GROUP_STATE_TOPIC_NAME)).allTopicNames().get().get(Topic.SHARE_GROUP_STATE_TOPIC_NAME)
-                .partitions().stream()
-                .filter(info -> info.partition() == shareGroupStateTp)
-                .map(info -> info.leader().id())
-                .toList();
+            // consume messages - start after small delay
+            ClientState consState = new ClientState();
+            // using map here if we want to debug specific keys
+            Map<String, Integer> consumed = new HashMap<>();
+            service.schedule(() -> {
+                    try (ShareConsumer<String, String> shareConsumer = createShareConsumer(groupId, Map.of(
+                        ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName(),
+                        ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName()
+                    ))) {
+                        shareConsumer.subscribe(List.of(topicName));
+                        while (!consState.done().get()) {
+                            ConsumerRecords<String, String> records = shareConsumer.poll(Duration.ofMillis(2000L));
+                            consState.count().addAndGet(records.count());
+                            records.forEach(rec -> consumed.compute(rec.key(), (k, v) -> v == null ? 1 : v + 1));
+                            if (prodState.done().get() && records.count() == 0) {
+                                consState.done().set(true);
+                            }
+                        }
+                    }
+                }, 100L, TimeUnit.MILLISECONDS
+            );
 
-            assertEquals(1, curShareCoordNodeId.size());
+            service.schedule(() -> {
```

Comment:
A comment above here please. Something like "Shut down the current share coordinator node after a short delay" just to aid with readability.